vnext/proposals/active/keyboard-reconcile-desktop.md

```diff
@@ -0,0 +1,148 @@
+
+- Deltas between implementations in order to reconcile one with the other
+- Deltas in scenarios that need to be completed in both
+- Any other fall out work
+
+## acceptsKeyboardFocus
+
+Property that determines whether this View, Touchable should be focusable with a non-touch input device, eg. receive focus with a hardware keyboard. Setting the property to true will make that View compoonent become a part of tabbing loop.
+
+| **TYPE** | **REQUIRED** | **PLATFORM** | **DEFAULT** |
+| --- | --- | --- | --- |
+| boolean | No | Windows, macOS | false |
+
+React Native core introduced `focusable` that does the same thing as acceptsKeyboardFocus in v0.62. In order to align and make this complete, the work here includes the following:
+- **COMPLETED** : Windows has added `focusable` and marked `acceptsKeyboardFocus` as deprecated in 0.62. 
+- macOS needs to do add `focusable` and mark `acceptsKeyboardFocus` as deprecated. Tracked by - [Issue#498](https://github.com/microsoft/react-native-macos/issues/498)
+- **COMPLETED** : Add focusable to Pressable ([windows Issue#5512](https://github.com/microsoft/react-native-windows/issues/5512), [macOS Issue#500](https://github.com/microsoft/react-native-macos/issues/500)). 
+- `focusable` is already supported in Android and exists in core. There is no hard upstreaming requirement, however – it would be good to implement this for iOS/iPadOS for completion.
+
+## focus, blur, onFocus, onBlur on View
+
+React Native core supports `.focus()`, `.blur()` methods and `onFocus()`, `onBlur()` callbacks on TextInput component. Windows and macOS implementations add the same methods and callbacks on View in addition to supporting them on TextInput. The work remaining for these include:
+- **COMPLETED** : The event bubbling behavior with .focus() and .blur() on View in macOS has some differences from Windows and w3c standards. The behavior needs to be evaluated and reconciled between the 2 platforms.
+- The microsoft/react-native fork has implementations for all the above methods and callbacks for iOS and Android as well. Once the event bubbling behavior above has been reconciled, these can be upstreamed to core. Tracked by [Issue#243](https://github.com/microsoft/react-native-macos/issues/243)
+- TextInput also includes an `isFocused()` method that returns true/false based on whether this TextInput currently has focus. It will be useful to add this to View to and upstream as a separate PR. Tee this up with the previous one. Tracked by - [windows Issue#5511](https://github.com/microsoft/react-native-windows/issues/5511), [macOS Issue#499](https://github.com/microsoft/react-native-macos/issues/499)
```

Comment:
I think we'd like to move away from methods on the ref in the future (including `focus` and `blur`) /cc @TheSavior. Should we explore either introducing a top-level API similar to the W3C `document.activeElement` or a function that accepts a ref to a host node and returns whether it is the focused element, e g., `isActiveElement(ref)` / `isFocused(ref)`? Or tackle all these changes to remove methods from refs all at once at a later date?

Reply:
We had to leave methods on the ref while we are in a world where both fabric and paper exist so we had to put the plans to move to a top-level api on hold. The issue is that `ref.blur` can change the implementation of `blur` based on which React renderer you are using. We don't have a way to make `import {blur} from 'react-native'; blur(ref);` use a different implementation of blur based on which renderer you are using.

Reply:
Per @TheSavior 's comments, leaving this as is for now to match what is in core. Can revisit on the whole post Fabric.